source/vistas/core/graphics/features.py

```diff
@@ -0,0 +1,356 @@
+        super().post_render(camera)
+
+
+class FeatureMesh(Mesh):
+
+    def __init__(self, vertices, indices):
+        super().__init__(len(indices), len(vertices), has_color_array=True, mode=Mesh.TRIANGLES)
+
+        color_buf = self.acquire_color_array()
+        color_buf[:] = numpy.ones_like(vertices.ravel()) * 0.5  # init the color buffer with grey
+        self.release_color_array()
+
+        self.shader = FeatureShaderProgram()
+        self.shader.feature = self
+
+
+class FeatureCollectionRenderThread(Thread):
+
+    def __init__(self, collection, scene):
+        super().__init__()
+        self.collection = collection
+        self.scene = scene
+        self.task = Task("Rendering feature collection")
+
+    def run(self):
+        self.init_event_loop()
+
+        if self.collection.needs_vertices:
+            self.task.status = Task.RUNNING
+            self.task.name = 'Generating meshes'
+            verts, indices = self.collection.generate_meshes(self.task)
+            self.sync_with_main(self.collection.add_features_to_scene, (verts, indices, self.scene), block=True)
+            self.collection.needs_vertices = False
+
+        if self.collection.needs_color:
+            self.task.status = Task.RUNNING
+            self.task.name = 'Coloring meshes'
+            colors = self.collection.generate_colors(self.task)
+            self.sync_with_main(self.collection.color_features, (colors, None), block=True)
+            self.collection.needs_color = False
+
+        self.sync_with_main(post_redisplay)
+        self.task.status = Task.COMPLETE
+
+
+class FeatureRenderable(Renderable):
+    def __init__(self, vertices, indices):
+        super().__init__()
+        self.mesh = FeatureMesh(vertices, indices)
+        self.bounding_box = self.mesh.bounding_box
+
+    def render(self, camera):
+        self.mesh.shader.pre_render(camera)
+        glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, self.mesh.index_buffer)
+        glDrawElements(self.mesh.mode, self.mesh.num_indices, GL_UNSIGNED_INT, None)
+        self.mesh.shader.post_render(camera)
+        glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, 0)
+
+    @property
+    def transparency(self):
+        return self.mesh.shader.alpha
+
+    @transparency.setter
+    def transparency(self, alpha):
+        """ Set the transparency of the feature layer. """
+
+        if alpha > 1.0:
+            alpha = 1.0
+        elif alpha < 0.0:
+            alpha = 0.0
+        self.mesh.shader.alpha = alpha
+
+    @property
+    def height_multiplier(self):
+        return self.mesh.shader.height_multiplier
+
+    @height_multiplier.setter
+    def height_multiplier(self, multiplier):
+        self.mesh.shader.height_multiplier = multiplier
+
+    @property
+    def height_offset(self):
+        return self.mesh.shader.height_offset
+
+    @height_offset.setter
+    def height_offset(self, offset):
+        self.mesh.shader.height_offset = offset
+
+
+class FeatureCollection:
```

Comment:
Yeah, `FeatureLayer` could work. My concern with `FeatureCollection` is that it already means something else. And "renderable" doesn't apply because this isn't that. So yeah, let's go with "layer".